indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/GeneratedPartitionsReport.java

```diff
@@ -37,11 +38,19 @@ public class GeneratedPartitionsReport implements SubTaskReport
   private final List<PartitionStat> partitionStats;
   private final Map<String, TaskReport> taskReport;
 
-  GeneratedPartitionsReport(String taskId, List<PartitionStat> partitionStats, Map<String, TaskReport> taskReport)
+  private final Long segmentsRead;
+
+  GeneratedPartitionsReport(
+      String taskId,
+      List<PartitionStat> partitionStats,
+      Map<String, TaskReport> taskReport,
+      Long segmentsRead
```

Comment:
It is weird to have this as a top-level field in all of these classes when we already have a well-structured `TaskReport`. Ideally, this information should be present inside the `taskReport` map passed to this class and then later be extracted and processed in `ParallelIndexSupervisorTask` as needed.

Reply:
The goal was to avoid adding unnecessary fields to the `IngestionStatsAndErrorReportData` where it would show up on reports everywhere